File: 0001-two-sum/0001-two-sum.py
class Solution:
    '''
    [3,3] 6 [0,1]

    [1,2,5,3,8] 4 [0,3]

    '''
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)

        seen = {}
        for i in range(n):
            diff = target - nums[i]
            if diff in seen:
                return [seen[diff], i]
            
            seen[nums[i]] = i
        return [-1, -1]
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        var seen = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            var diff = target - nums[i];
            if (seen.containsKey(diff)) {
                return new int[]{seen.get(diff), i};
            }

            seen.put(nums[i], i);
        }
        
        return new int[]{-1, -1};
    }
}